src/pudl/transform/vceregen.py

```diff
@@ -0,0 +1,257 @@
+        == len(cap_fac_dict["offshore_wind"])
+        == len(cap_fac_dict["onshore_wind"])
+        == len(mega_df)
+    ):
+        raise AssertionError("Merge issue causing a length difference.")
+
+    return mega_df
+
+
+def _combine_cap_fac_with_fips_df(
+    cap_fac_df: pd.DataFrame, fips_df: pd.DataFrame
+) -> pd.DataFrame:
+    """Combine the combined capacity factor df with the fips df."""
+    logger.info(
+        "Merging the combined capacity factor table with the Lat-Long-FIPS table"
+    )
+    combined_df = pd.merge(cap_fac_df, fips_df, on="county_state_names", how="left")
+    if len(combined_df) != len(cap_fac_df):
+        raise AssertionError("Merge erroniously altered table length.")
+    return combined_df
+
+
+@asset(
+    io_manager_key="parquet_io_manager",
+    compute_kind="pandas",
+    op_tags={"memory-use": "high"},
+)
+def out_vceregen__hourly_available_capacity_factor(
+    raw_vcegen__lat_lon_fips: pd.DataFrame,
+    raw_vceregen__fixed_solar_pv_lat_upv: pd.DataFrame,
+    raw_vceregen__offshore_wind_power_140m: pd.DataFrame,
+    raw_vceregen__onshore_wind_power_100m: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform raw Vibrant Clean Energy renewable generation profiles.
+
+    Concatenates the solar and wind capacity factors into a single table and turns
+    the columns for each county into a single county column.
+
+    Note that this transform is a bit unusual, in that it is producing a highly
+    processed output table. That's because we're working backwards from an archived
+    finished product to be able to provide a minimum viable product. Our intent is to
+    integrate or reimplement the steps required to produce this output table from
+    less processed original inputs in the future.
+    """
+    logger.info("Transforming the hourly available capacity factor tables")
+    # Clean up the FIPS table
+    fips_df = _prep_lat_long_fips_df(raw_vcegen__lat_lon_fips)
+    # Apply the same transforms to all the capacity factor tables. This is slower
+    # than doing it to a concatinated table but less memory intensive because
+    # it doesn't need to process the ginormous table all at once.
+    df_dict = {
+        "solar_pv": raw_vceregen__fixed_solar_pv_lat_upv,
+        "offshore_wind": raw_vceregen__offshore_wind_power_140m,
+        "onshore_wind": raw_vceregen__onshore_wind_power_100m,
+    }
+    for df_name, df in df_dict.items():
+        logger.info(f"Prepping the {df_name} table")
+        df_dict[df_name] = (
+            _check_for_valid_counties(df, fips_df)
+            .pipe(_make_cap_fac_frac, df_name)
+            .pipe(_add_time_cols)
+            .pipe(_stack_cap_fac_df, df_name)
+        )
+    # Make sure there's no funny business
+    if len(df_dict) != 3:
+        raise AssertionError("Incorrect number of dataframes in the df_dict")
+    # Combine the data!
+    out_df = _combine_all_cap_fac_dfs(df_dict).pipe(
+        _combine_cap_fac_with_fips_df, fips_df
```

Comment:
Tempted to try combining each of the cap fac tables with fips in the for loop above instead of here to see if that would save any computing power.